### PR TITLE
[Merged by Bors] - Reduced the size of AST nodes

### PR DESCRIPTION
**Pull request by Razican**
This PR boxes the `Try` and `TaggedTemplate` nodes reducing the size of the `Node` structure from 88 to 56 bytes. This should improve performance in most cases, by adding another indirection with code with `try` and templates.

### CODE REVIEW

**Comment by Razican** [comment]
I'm actually pretty surprised that the benchmark are not clearly showing improvements, even though there might be some small ones. There is also one benchmark that shows a 18% worsening, which is worrying me. Maybe this is just due to the GitHub Actions uncertainty (no dedicated processing power, for instance).

We should probably try to benchmark things locally or in a dedicated-core virtual server.

**Comment by jasonwilliams** [comment]
@Razican have you tried running the profiler and looking at before and after? That may give you a deeper insight into what is going on.

https://github.com/boa-dev/boa/blob/main/docs/profiling.md

**Comment by Razican** [comment]
> @Razican have you tried running the profiler and looking at before and after? That may give you a deeper insight into what is going on.
> 
> https://github.com/boa-dev/boa/blob/main/docs/profiling.md

I ran the profiler locally and didn't see much difference. Then I ran benchmarks, and I could clearly see a consistent 2-6% improvement in the parser and a consistent and clear 1-2% improvement in the execution. The compiler was having a 0-1% improvement, but it was not very clear.

I also checked this "Array creation (Execution)" benchmark and got a 1.8% improvement locally, so that 18% worsening is definitely not real.

File: boa_wasm/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ rust-version = "1.56"
 
 [dependencies]
 Boa = { path = "../boa", features = ["console"] }
-wasm-bindgen = "=0.2.78"
+wasm-bindgen = "=   0.2.78"
```
Comment (RageKnify) [inline boa_wasm/Cargo.toml:16]: What does this do?

**Review by RageKnify — APPROVED** [review]
Confused about the change in `boa_wasm/Cargo.toml`, the rest looks fine.

File: boa_wasm/Cargo.toml (hunk 1 above)
Comment (Razican) [reply boa_wasm/Cargo.toml]: It was a typo, sorry, it has been fixed :)

**Review by HalidOdat — APPROVED** [review]

**Pull request closed without merging**